File: lidar_msgs/scripts/lidar_msgs.py
#!/usr/bin/env python3

"""
chmod +x scripts/lidar_msgs.py
roslaunch wpr_simulation wpb_simple.launch
rosrun lidar_msgs lidar_msgs.py
"""
import rospy
from sensor_msgs.msg import LaserScan

def lidar_callback(msg: LaserScan):
    rospy.loginfo(f'DistanceToForward: {msg.ranges[180]}')

if __name__ == "__main__":
    rospy.init_node("lidar_msgs")
    rospy.Subscriber("/scan", LaserScan, queue_size=3, callback=lidar_callback)
    
    rospy.spin()

File: lidar_msgs/src/lidar_msgs.cpp
/*
roslaunch wpr_simulation wpb_simple.launch
rosrun lidar_msgs lidar_msgs
*/

#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>

void lidar_callback(sensor_msgs::LaserScan msg)
{
    float distanceToForward = msg.ranges[180];
    ROS_INFO_STREAM("distance to forward:" << std::to_string(distanceToForward) << "\n");
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "lidar_msg");

    ros::NodeHandle nh;

    ros::Subscriber subscriber = nh.subscribe<sensor_msgs::LaserScan>("/scan", 3, lidar_callback);

    ros::Rate rate = ros::Rate(10);

    while(ros::ok())
    {
        ros::spinOnce();
        rate.sleep();
    }
    /* code */
    return 0;
}
